apps/arweave/src/ar_data_sync.erl

```diff
@@ -3422,6 +3453,37 @@ get_merkle_rebase_threshold() ->
 			Threshold
 	end.
 
+%% @doc Update the replica 2.9 entropy-aligned record of the synced chunks.
+%% It differs from the normal record (ar_data_sync) in that it only registers
+%% the bucket numbers of the synced chunks and records chunks with the same footprint
+%% next to each other. For example, a record may contain intervals 0-10, 1000-1024,
+%% 1020-2048. This means the node has the first 10 chunks of the first entropy footprint,
+%% the last 24 chunks of the first entropy footprint and chunks 20-48 of the second
+%% entropy footprint. These chunks are from the first partition. The offset of the chunks
+%% from the second partition is shifted by the number of chunks in the replica 2.9
+%% entropy generated per partition (which is slightly bigger than the number of chunks
+%% that can fit in the 3.6 TB partition).
+%%
+%% Note that Packing does not have to be replica_2_9. We maintain this record
+%% for any packing so that it is convenient to serve the data to any client.
+add_footprint(PaddedOffset, Packing, StoreID) ->
```

Comment:
`add_footprint` is adding a single chunk to the footprint record? Or its adding a full footprint?

Reply:
A single chunk